client-v2/src/services/editionsApi.ts

```diff
@@ -0,0 +1,57 @@
+import { EditionIssue } from 'types/Edition';
+import { Moment } from 'moment';
+import pandaFetch from './pandaFetch';
+
+const dateFormat = 'YYYY-MM-DD';
+
+export const fetchIssuesForDateRange = async (
+  start: Moment,
+  end: Moment
+): Promise<EditionIssue[]> => {
+  return pandaFetch(
+    `http://localhost:3000/editions-api/issues?start=${start.format(
+      dateFormat
+    )}&end:${end.format(dateFormat)}`,
+    {
+      method: 'get',
+      credentials: 'same-origin'
+    }
+  ).then(response => response.json());
+};
+
+export const fetchIssueByDate = async (date: Moment): Promise<EditionIssue> => {
+  return pandaFetch(
+    `http://localhost:3000/editions-api/issues/${date.format(dateFormat)}`,
+    {
+      method: 'get',
+      credentials: 'same-origin'
+    }
+  )
+    .then(response => {
+      if (response.status === 200) {
+        return response.json();
+      } else {
+        return [];
```

Comment:
Is this else statement needed? Did we determine that the catch would be fired if the if statement fails?

Reply:
Now that I look at this code, I'm quite surprised that this compiles (possibly pandaFetch's typing isn't too strict) -- it feels like the return type should be `Promise<EditionIssue|null>` or similar. We definitely don't need the else, and we could choose to explicitly return null after the if condition,  or simply write nothing there to return `undefined`.